tests/docs/test_complete_examples.py

```diff
@@ -0,0 +1,226 @@
+import csv
+import inspect
+import typing
+import unittest.mock
+import uuid
+import warnings
+from collections.abc import Generator, Iterator
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import pytest
+
+import ehrql.main
+
+
+@dataclass
+class DatasetDefinitionExample:
+    """Store details of a complete ehrQL dataset definition example."""
+
+    path: Path
+
+    start_line: int
+    end_line: int | None = None
+
+    content_lines: list[str] = field(default_factory=list)
+
+    def relative_path(self) -> Path:
+        """Return the relative path of the dataset definition source file
+        to the source code root."""
+        source_code_path = Path(__file__).parents[2]
+        return self.path.relative_to(source_code_path)
+
+
+def discover_paths(glob_string: str) -> Generator[Path, None, None]:
+    """Generate a list of matching files for a glob in the documentation source path."""
+    docs_path = Path(__file__).parents[2] / "docs"
+    return docs_path.glob(glob_string)
+
+
+def find_complete_ehrql_examples_in_markdown(
+    file: typing.TextIO,
+) -> Iterator[DatasetDefinitionExample]:
+    """Yields extracted code blocks labelled as ```ehrql from a Markdown file.
+
+    Incomplete ehrQL dataset definitions should be labelled as ```python,
+    and not with ```ehrql."""
+    # There are three kinds of possible ehrql code blocks that we might see.
```

Comment:
[This isn't true](https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#nested-fence-format), I'll have a go at correcting this.